public/js/mage/adminhtml/tools.js

```diff
@@ -754,3 +755,74 @@ function copyText(event) {
         copyIcon.classList.remove('icon-copy-copied');
     }, 1000);
 }
+
+/**
+ * Clear <div id="messages"></div>
+ */
+function clearMessageDiv() {
+    setMessageDivHtml('');
+}
+
+/**
+ * Set a message in <div id="messages"></div>
+ *
+ * @param {string} message - text value of the message to display
+ * @param {string} type - one of `success|error|notice`
+ */
+function setMessageDiv(message, type = 'success') {
+    setMessageDivHtml(`<ul class="messages"><li class="${type}-msg"><ul><li><span>${message}</span></li></ul></li></ul>`);
+}
+
+/**
+ * Raw function to update <div id="messages"></div>
+ *
+ * @param {string} html
+*/
+function setMessageDivHtml(html) {
+    const div = document.getElementById('messages');
+    if (div) {
+        div.innerHTML = html;
+    }
+}
+
+/**
+ * Set Varien type route params, i.e. /id/1/
+ *
+ * @param {string} url - the base URL
+ * @param {Object} params - key value pairs to add, update, or remove
+ */
+function setRouteParams(url, params = {}) {
+    url = new URL(url);
+    if (!url.pathname.endsWith('/')) {
+        url.pathname += '/';
+    }
+    for (const [ key, val ] of Object.entries(params)) {
+        const regex = new RegExp(String.raw`\/${key}\/\w+\/`);
+        if (val === null || val === false) {
+            url.pathname = url.pathname.replace(regex, '/');
+        } else if (url.pathname.match(regex)) {
+            url.pathname = url.pathname.replace(regex, `/${key}/${val}/`);
+        } else {
+            url.pathname += `${key}/${val}/`;
+        }
+    }
+    return url.toString();
+}
+
+/**
+ * Set query params, i.e. ?id=1
+ *
+ * @param {string} url - the base URL
+ * @param {Object} params - key value pairs to add, update, or remove
+ */
+function setQueryParams(url, params = {}) {
```

Comment:
would it make sense to you to prefix all new functions with `maho`? The thing is, when looking for a maho-javascript-something in an IDE, if we have a prefix, it's much easier, and we also avoid possible name collisions?

Reply:
Created #84 for us to discuss 😃